src/pipeline/agenticRag/run.py

```diff
@@ -389,4 +389,6 @@ async def run(self, clinical_info: Any, max_retries: int = 3) -> Dict[str, Any]:
         self.logger.error(
             f"{self.prefix}Max retries reached. Returning empty list of policies."
         )
-        return {"query": None, "policies": [], "evaluation": None}
+
+        # this was updated to return information about the evaluation, even in failure we need to have the reasoning.
+        return {"query": expanded_query, "policies": evaluation.get("policies", []), "evaluation": evaluation}
```

Comment:
@pablosalvador10 @JinLee794 will this change break anything? When the retry conditional hit, we're omitting the reasoning. I would like to access this for evaluations.